enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,584 @@
+
+This should make no noticable difference on "regular" single control-plane node
+clusters which do not have any day-1 or day-2 worker nodes. The only difference
+for those clusters would be the `IngressController` targeting the "master" pool
+rather than the "worker" pool, but since the single control-plane node is
+already both in the "master" and "worker" pools, that should make no practical
+difference.
+
+On multi-node clusters, the installer will never set the
+`DefaultPlacement` field to `ControlPlane`, so there are no risks to
+discuss for multi-node clusters in this enhancement. However, any future
+enhancement that will consider making this field configurable by allowing
+the user to set it during installation, should take into account that the
+documentation / installation process for the load balancers would have to take
+this field into account when choosing which nodes the load balancer should
+target.
+
+I do not believe this enhancement has any security implications.
+
+## Design Details
+
+### Open Questions
+
+None that I can think of at the moment
+
+### Test Plan
+
+- Add unit tests to the Cluster Ingress Operator to make sure the
+IngressController resource is generated as exepected.
+
+- Add periodic nightly tests which install single-node in the cloud, add a few
+worker nodes to it, then run conformance tests to make sure we don't run into
+any problems not described in this enhancement.
+
+- Add periodic nightly tests which install a single-node "none"-platform cluster,
+add worker nodes to it, and check that ingress traffic still works as expected
+and recovers even after the `router-default` pod gets deleted and rescheduled.
+Make sure this is still true even after upgrades.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after the single control-plane node reboots by
+running conformance tests post-reboot.
+
+- Add tests on both cloud / "none"-platform that check that a single-node cluster
+with additional workers recovers after an upgrade by running conformance tests
+post-upgrade.
+
+TODO: Describe day-1-workers tests?
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+The new `DefaultPlacement` API field will have a default value of `Workers`.
+The behavior of the Cluster Ingress Operator when the `DefaultPlacement` field
+has the value `Workers` should be identical to its current behavior (before
+this enhancement). This means that clusters that go through an upgrade from a
+version before this enhancement to a later version which includes this
+enhancement will maintain their current behavior, this enhancement should not
+change anything for them.
+
+If the value of `DefaultPlacement` is empty (TODO: Could this possibly happen
```

Comment:
@omertuc This TODO item looks like an open question. Is it answered elsewhere?

Reply:
It is not answered elsewhere, however, my understanding of how CRDs work has improved and I'm pretty certain of the answer, so I'll remove the TODO